### PR TITLE
test/ctr: fix "should not leak children" test case

### PR DESCRIPTION
**Pull request by kolyshkin**
_Based on (and currently includes) #6950; draft until that one is merged._

    
The test was getting ALL conmon PIDS, which doesn't really work in case
we run tests in parallel (which we do), or in case conmon-rs is used.
    
In fact, what we need is PIDs of all processes in the container. The
best place to get these is cgroups.
    
Do that, and simplify process status check using grep.
    
Fixes: 9fea6e219e
Fixes: #6979 

/kind ci

```release-note
NONE
```

### CODE REVIEW

File: test/ctr.bats [hunk 1]
```diff
@@ -1006,11 +997,18 @@ function check_oci_annotation() {
...
 
 	# make sure crio syncs state
-	for process in ${processes}; do
-		# Ignore Z state (zombies) as the process has just been killed and reparented. Systemd will get to it.
-		# `pgrep` doesn't have a good mechanism for ignoring Z state, but including all others, so:
-		# shellcheck disable=SC2143
-		[ -z "$(ps -p "$process" o pid=,stat= | grep -v ' Z')" ]
+	for pid in ${processes}; do
+		if kill -0 "$pid"; then
+			# The proces is still there. If it's a zombie,
```
Comment (haircommander) [inline test/ctr.bats:1002]: process*

Comment (kolyshkin) [reply test/ctr.bats]: fixed

**Comment by kolyshkin** [comment]
Test failures ("not ok 182 metrics container oom" and "not ok 291 emit metric when container is re-requested") are not related and are being addressed elsewhere (#6973 and #7019).

**Pull request closed without merging**